Added first set of SQL docs (#290)
Macrometacorp/docs: docs/sql/index.md

---
sidebar_position: 1
title: SQL
---

> # SQL Query Language

Reviewer: ```suggestion
```

Reviewer: SQL = Structured Query Language

The title is redundant, let's remove it.

> 

Reviewer: Is this an alpha or beta feature? If so, then we need to note it at the top and let people know what to expect.

> SQL for Structured Query Language. The SQL query language can be used to create, retrieve and modify data that are stored in the Macrometa geo-distributed fast data platform.

Reviewer: ```suggestion
_SQL_ stands for Structured Query Language. You can use SQL to create, retrieve, and modify data that are stored in the Macrometa geo-distributed fast data platform.
```

:::note

> If this is your first time with SQL then be sure to check out the [SQL GoT Tutorial](got-tutorial/index.md) before you head off to the in-depth documentation!

Reviewer: Be sure to update this link and text.

:::

> Structured Query Language (SQL) is a standardized programming language that is used to manage relational databases and perform various operations on the data in them.

Reviewer: Didn't we already cover this in line 8?

Although SQL is an ANSI/ISO standard, there are different versions of the SQL language. Macrometa SQL dialect supports DQL (SELECT statement), DML statements (UPDATE, INSERT and DELETE) and a subset of DDL statements.

> The general workflow when executing a query is as follows:

Reviewer: Love this, but I suggest putting a heading above it. Something like:

## SQL Workflow

- A client application sends a SQL query to the Macrometa platform. The query text contains everything Macrometa needs to compile the result set.

> 

Reviewer: ```suggestion
```

> - Macrometa will parse the query, execute it and compile the results. If the query is invalid or cannot be executed, the server will return an error that the client can process and react to. If the query can be executed successfully, the server will return the query results (if any) to the client.

Reviewer: ```suggestion
- Macrometa parses the query, executes it, and compiles the results. If the query is invalid or cannot be executed, then the server returns an error that the client can process and react to. If the query can be executed successfully, then the server will return the query results (if any) to the client.
```

Reviewer: Great stuff, here. I just want to explain a few of my edits:

- Try to write in present tense. Any time you see "will," take a look at the verb and see if you can change it from future to present tense. In this case, will parse to parses. This is clearer and removes an unnecessary word.
- Always use the serial comma, which is the comma before an "and."
- Unless there is a reason not to, use "then" when you start the sentence with "if." If-then statements are a great pattern, and the signposts help less-practiced English speakers understand the text.

Last comment - try not to use parentheticals in sentences. They can be confusing to readers.
